website/src/analytics.js

```diff
@@ -0,0 +1,124 @@
+var _ = require('lodash');
+var Amplitude = require('amplitude');
+var googleAnalytics = require('universal-analytics');
+
+var ga;
+var amplitude;
+
+var analytics = {
+  trackPurchase: trackPurchase,
+  track: track
+}
+
+function init(options) {
+  if(!options) { throw 'No options provided' }
+
+  amplitude = new Amplitude(options.amplitudeToken);
+  ga = googleAnalytics(options.googleAnalytics);
+
+  return analytics;
+}
+
+function track(eventType, data) {
+  _sendDataToAmplitude(eventType, data);
+  _sendDataToGoogle(eventType, data);
+}
+
+function _sendDataToAmplitude(eventType, data) {
+  var amplitudeData = _formatDataForAmplitude(data);
+  amplitudeData.event_type = eventType;
+  amplitude.track(amplitudeData);
+}
+
+function _sendDataToGoogle(eventType, data) {
+  var category = data.category;
+  var label = _generateLabelForGoogleAnalytics(data);
+
+  ga.event(category, eventType, label).send();
+}
+
+function _generateLabelForGoogleAnalytics(data) {
+  var label = 'Label Not Specified';
+  var POSSIBLE_LABELS = ['gaLabel', 'itemName', 'gemCost', 'goldCost'];
+
+  _(POSSIBLE_LABELS).each(function(key) {
+    if(data[key]) {
+      label = data[key];
+      return false; // exit _.each early
+    }
+  });
+
+  return label;
+}
+
+function trackPurchase(data) {
+  _sendPurchaseDataToAmplitude(data);
+  _sendPurchaseDataToGoogle(data);
+}
+
+function _sendPurchaseDataToAmplitude(data) {
+  var amplitudeData = _formatDataForAmplitude(data);
+  amplitudeData.event_type = 'purchase';
+  amplitudeData.revenue = data.purchaseValue;
+
+  amplitude.track(amplitudeData)
+}
+
+function _formatDataForAmplitude(data) {
+  var PROPERTIES_TO_SCRUB = ['uuid', 'user', 'purchaseValue', 'gaLabel'];
+  var event_properties = _.omit(data, PROPERTIES_TO_SCRUB);
```

Comment:
Ah, this is nice. We could do something like this on the client side to remove some of the GA noise from Amplitude events.